stl/inc/syncstream

```diff
@@ -0,0 +1,312 @@
+// syncstream standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SYNCSTREAM_
+#define _SYNCSTREAM_
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#ifdef _M_CEE_PURE
+#error <syncstream> is not supported when compiling with /clr:pure
+#endif // _M_CEE_PURE
+
+#ifndef __cpp_lib_syncbuf
+#pragma message("The contents of <syncstream> are available only with C++20 syncbuf support.")
+#else // ^^^ !defined(__cpp_lib_syncbuf) / defined(__cpp_lib_syncbuf) vvv
+#include <map>
+#include <memory>
+#include <shared_mutex>
+#include <streambuf>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#pragma push_macro("emit")
+#undef new
+#undef emit
+
+_STD_BEGIN
+
+struct _Mutex_count_pair {
+    shared_mutex _Mutex;
+    uint64_t _Ref_count = 0;
+};
+
+_CRTIMP2_PURE map<void*, _Mutex_count_pair>& __cdecl _Get_mutex_map();
+_CRTIMP2_PURE shared_mutex& __cdecl _Get_map_mutex();
+
+shared_mutex& _Get_mutex_for_instance(void* _Ptr) {

```

Comment:
These functions need to either be made `inline` or moved to a DLL. Since we need to use DLL anyway it's probably better to move these functions there.

Reply:
I'll move these functions to `syncstream.cpp`, it seems the better solution like you said.